ha/core/event_manager/event_manager.py

```diff
@@ -57,42 +89,181 @@ def _validate_component(self, component: str) -> None:
+                # Remove the event from the list(coming from consul)
+                if delete_required_state in event_list:
+                    event_list.remove(delete_required_state)
+                if event_list:
+                    # After deletion, if list is not empty, update the consul key
+                    new_event_list = json.dumps(event_list)
+                    self._confstore.update(f'{const.COMPONENT_KEY}/{component}', new_event_list)
+                else:
+                    # else, delete the whole component
+                    self._confstore.delete(f'{const.COMPONENT_KEY}/{component}')
+            else:
+                # If component key is not there, means event with that key will not be there,
+                # hence safely raise exception here so that _delete_event_key will not be called
+                raise UnSubscribeException('Can not unsubscribe the component: \
+                                            {component} as it was not regostered earlier')
+
+    def _delete_event_key(self, component: str, resource_type: str, states: list = None):
+        '''
+          Deletes the event from the component key
+          Ex:
+          Already existed consul key: cortx/ha/v1/events/node:os:memory_usage/failed:["motr", "hare"]
+          Request to delete: 'hare', 'node:os:memory_usage' 'failed'
+          Stored key after deletion will be: cortx/ha/v1/events/node:os:memory_usage/failed:["motr"]
+        '''
+        for state in states:
+            if self._confstore.key_exists(f'{const.EVENT_KEY}/{resource_type}/{state}'):
+                comp_json_list = self._confstore.get(f'{const.EVENT_KEY}/{resource_type}/{state}')
+                # Get the component list
+                comp_list = json.loads(comp_json_list['cortx/ha/v1/' + f'{const.EVENT_KEY}/{resource_type}/{state}'])
+                # Remove component from the list
+                if component in comp_list:
+                    comp_list.remove(component)
+                if comp_list:
+                    # If still list is not empty, update the consul key
+                    new_comp_list = json.dumps(comp_list)
+                    self._confstore.update(f'{const.EVENT_KEY}/{resource_type}/{state}', new_comp_list)
+                # Else delete the event from the consul
+                else: self._confstore.delete(f'{const.EVENT_KEY}/{resource_type}/{state}')
+            else:
+                print('Error')
+
+    def unsubscribe(self, component : str, events : list = None):
+        '''
+           Unregister the event for the specific component and the component \
+           for the event using consul deletion
+           Args:
             component (str): [description]
             events (list, optional): [description]. Defaults to None.
-
-        Raise:
+           Raise:
             UnSubscribeException: Raise error if failed.
-        """
-        pass
+        '''
+        self._validate_events(events)
```

Comment:
Both component and events could be None or One of them could be None as well. Please handle this case here and in subscribe as well.

Reply:
Done